### PR TITLE
17216 - Show start date for a firm dissolution filing 

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#17216

*Description of changes:*

Show start date for a firm dissolution filing

Business start date in Business info section:
image

Before:
image

After:
image

Note: Dissolution date rules do not need to be updated. It is working as expected. Confirmed with Mihai.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-568-fcw9wifm.web.app

File: src/interfaces/store-interfaces/state-interfaces/business-interface.ts [hunk 1]
```diff
@@ -38,5 +38,6 @@ export interface BusinessIF {
   taxId?: string // aka Business Number // may be undefined
   state: EntityState
   stateFiling?: string
+  startDate: ApiDateTimeUtc | string
```
Comment (severinbeauvais) [inline src/interfaces/store-interfaces/state-interfaces/business-interface.ts:41]: I believe `ApiDateTimeUtc` is an alias for string, so you shouldn't need the union type.

File: src/mixins/filing-template-mixin.ts [hunk 2]
```diff
@@ -829,6 +831,7 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
     this.setEntityType(draftFiling.business.legalType || this.getBusinessLegalType)
     this.setLegalName(draftFiling.business.legalName || this.getBusinessLegalName)
     this.setFoundingDate(draftFiling.business.foundingDate || this.getBusinessFoundingDate)
+    this.setBusinessStartDate(this.getBusinessStartDate)
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:834]: Why not save it to, and restore it from, `draftFiling.business.startDate` (same as other business properties)?

Comment (ketaki-deodhar) [reply src/mixins/filing-template-mixin.ts:834]: `startDate` is not returned in the response of `fetchFirstTask` legal api call. Hence, setting it in business Info

image

image

Comment (severinbeauvais) [reply src/mixins/filing-template-mixin.ts:834]: Ah, I see.

This "fetch first task" thing is kludgey anyway. I would be better to pass in the id of the filing, but that's a future problem.

Wouldn't it be much cleaner to update Filings UI so it populates Start Date in the draft?

Comment (ketaki-deodhar) [reply src/mixins/filing-template-mixin.ts:834]: Yes. I will change Edit UI and create a PR soon

File: src/interfaces/filing-interfaces/filing-interfaces.ts [hunk 3]
```diff
@@ -140,6 +140,7 @@ export interface DissolutionFilingIF {
     identifier: string
     legalName: string
     foundingDate: string
+    startDate: string
```
Comment (severinbeauvais) [inline src/interfaces/filing-interfaces/filing-interfaces.ts:143]: Is this only non-null for firms? Might be worth a comment.

File: src/mixins/filing-template-mixin.ts [hunk 4]
```diff
@@ -831,7 +832,7 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
     this.setEntityType(draftFiling.business.legalType || this.getBusinessLegalType)
     this.setLegalName(draftFiling.business.legalName || this.getBusinessLegalName)
     this.setFoundingDate(draftFiling.business.foundingDate || this.getBusinessFoundingDate)
-    this.setBusinessStartDate(this.getBusinessStartDate)
+    this.setBusinessStartDate(draftFiling.business.startDate || this.getBusinessStartDate)
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:835]: This is a good fallback.

This means you still need to save business start date in App.vue, right?

Comment (ketaki-deodhar) [reply src/mixins/filing-template-mixin.ts:835]: Yes!

Comment (severinbeauvais) [reply src/mixins/filing-template-mixin.ts:835]: OK. Your code in App.vue fits in beautifully with the other code 👍

File: src/mixins/filing-template-mixin.ts (hunk 2 above)
Comment (severinbeauvais) [reply src/mixins/filing-template-mixin.ts]: This is resolved. I assume you'll test this and it will be fine :)

**Review by severinbeauvais — APPROVED** [review]
LGTM

**Review by chenhongjing — APPROVED** [review]
LGTM 🐱

**Pull request merged**